apps/vs-code-designer/src/app/utils/packageManagers/getPackageManager.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Platform } from '../../../constants';
+import { localize } from '../../../localize';
+import { executeCommand } from '../funcCoreTools/cpUtils';
+import { getWorkspaceSetting, updateGlobalSetting } from '../vsCodeConfig/settings';
+import { DialogResponses, callWithTelemetryAndErrorHandling } from '@microsoft/vscode-azext-utils';
+import type { IActionContext } from '@microsoft/vscode-azext-utils';
+import type { MessageItem } from 'vscode';
+
+export async function getPackageManager(): Promise<void> {
+  await callWithTelemetryAndErrorHandling('azureLogicAppsStandard.getPackageManager', async (context: IActionContext) => {
+    context.errorHandling.suppressDisplay = true;
+    context.telemetry.properties.isActivationEvent = 'true';
+    context.telemetry.properties.platform = process.platform;
+
+    const showMissingPackageManagerWarningKey = 'showMissingPackageManagerWarning';
+    const platformPackageManagerKey = 'PlatformPackageManager';
+
+    switch (process.platform) {
```

Comment:
Do we need to put this on a try catch? If the executeCommand function fails and throws an Error do it goes to the default case?

Reply:
I notice that this gets executed in the main file, without checking if the user has the package manager installed, thats why I came up with this scenario